Gopkg.toml

```diff
@@ -0,0 +1,57 @@
+
+# Gopkg.toml example
+#
+# Refer to https://github.com/golang/dep/blob/master/docs/Gopkg.toml.md
+# for detailed Gopkg.toml documentation.
+#
+# required = ["github.com/user/thing/cmd/thing"]
+# ignored = ["github.com/user/project/pkgX", "bitbucket.org/user/project/pkgA/pkgY"]
+#
+# [[constraint]]
+#   name = "github.com/user/project"
+#   version = "1.0.0"
+#
+# [[constraint]]
+#   name = "github.com/user/project2"
+#   branch = "dev"
+#   source = "github.com/myfork/project2"
+#
+# [[override]]
+#  name = "github.com/x/y"
+#  version = "2.4.0"
+
+required = [
+  "k8s.io/api/core/v1",
+]
+
+[[constraint]]
+  name = "k8s.io/api"
+  version = "kubernetes-1.8.5"
```

Comment:
We said we want to import all the necessary deps and pin them because dep does not inherit deps like glide did. Check kvm-operator for instance. This might not yet be a problem but I can see this being an issue as soon as you start working with the operator in an installation.

Reply:
There's probably something I don't understand, but do we want to pin the deps or not? Here: https://github.com/giantswarm/kvm-operator/blob/master/Gopkg.toml kvm-operator doesn't pin the versions. But the above referenced piece does. I added it after getting compilation errors and search in Slack brought a discussion which stated that said piece should be used (I think it was Pawel mentioning this ~2 weeks ago).

Reply:
Oh wow. I see. This is odd. I had lots of trouble with cluster-service and though kvm-operator is sorted already. See https://github.com/giantswarm/cluster-service/blob/master/Gopkg.toml. I found it was safer for updating dependencies using dep when you are explicit. The transition was just not possible without the pins in the cluster service so there was a conclusion also announced in #news-dev that we should always be as explicit as possible.